can/io/gz.py

```diff
@@ -0,0 +1,37 @@
+"""
+This module contains handling for .gz (gzip) files.
+"""
+
+import gzip
+import pathlib
+import tempfile
+
+from ..typechecking import AcceptedIOType
+
+
+class GzipReader:
+    """
+    Handle any supported file that may be compressed using gzip.
+    Decompresses the file and transfers contents to a tempfile.
+    The tempfile then gets re-consumed by LogReader and automatically
+    removed once complete.
+    """
+
+    @staticmethod
+    def __new__(cls, file: AcceptedIOType):
+        """
+        :param file: a path-like object or as file-like object to read from.
+        """
+        # Needed to avoid circular import error
+        from .player import LogReader
+
+        real_suffix = pathlib.Path(file).suffixes[-2]
+        file_mode = "b" if real_suffix in [".blf"] else "t"
+
+        with tempfile.NamedTemporaryFile(
+            mode=f"w{file_mode}", suffix=real_suffix
+        ) as log:
+            with gzip.open(file, f"r{file_mode}") as f:
+                log.write(f.read())
+
+            return LogReader(log.name)
```

Comment:
I think that after this return, the temporary file will get closed by *might* be removed.

Instead, we could make this class inherit from `MessageReader`, which inherits from `BaseIOHandler` which has a `stop()` method in which we can close resources. But then, we would have to abandon the `__new__` approach.

Reply:
It would be nice to be able to pass a file name *and* a an optional file-pointer to `LogReader`, one to resolve the reader type and one for the actual content. That way, we could read the file "lazily" by directly passing the `gzip.open()` result and avoid the temporary file. It would even get closed automatically by the specific reader.

Reply:
I like this idea as well and it seems the most logical. 

However if we're going to alter `LogReader`, I  think it makes more sense for `GzipReader` to become a method within `LogReader` since technically it's not _reading_ anything. This would also avoid the circular import issue which seems like a bit of a 'code  smell' imo.

Reply:
That sounds pretty simple now that you say it!

Reply:
Added! Would love feedback before I work on the tests.